README.md

```diff
@@ -0,0 +1,779 @@
+    transformation={[{ height: 200, width: 200, q: 50 }]}
+    controls={true}
+  />
+</ImageKitProvider>
+```
+
+#### Image lazy loading and low-quality placeholder
+```js
+<ImageKitProvider urlEndpoint="https://ik.imagekit.io/your_imagekit_id">
+  // Lazy loading image
+  <IKImage
+    path="/default-image.jpg"
+    transformation={[{
+      "height": "200",
+      "width": "200"
+    }]}
+    loading="lazy"
+    alt="Alt text"
+  />
+
+  /*
+     Low-quality image placeholder
+     Will first load https://ik.imagekit.io/your_imagekit_id/tr:h-200,w-200:q-20,bl-6/default-image.jpg, while the original image, i.e., https://ik.imagekit.io/your_imagekit_id/tr:h-200,w-200/default-image.jpg is being loaded in the background.
+  */
+  <IKImage
+    path="/default-image.jpg"
+    transformation={[{
+      "height": "200",
+      "width": "200"
+    }]}
+    lqip={{ active: true }}
+    alt="Alt text"
+  />
+
+  // Low-quality image placeholder with custom quality and blur values
+  <IKImage
+    path="/default-image.jpg"
+    transformation={[{
+      "height": "200",
+      "width": "200"
+    }]}
+    lqip={{ active: true, quality: 20, blur: 10 }}
+    alt="Alt text"
+  />
+
+  // Low-quality image placeholder and lazy loading of original image in the background
+  <IKImage
+    path="/default-image.jpg"
+    transformation={[{
+      "height": "200",
+      "width": "200"
+    }]}
+    loading="lazy"
+    lqip={{ active: true }}
+    alt="Alt text"
+  />
+</ImageKitProvider>
+```
+
+#### File upload example
+```js
+// Ensure you pass publicKey, urlEndpoint, and authenticator function to the parent ImageKitProvider component or to the IKUpload component directly.
+<ImageKitProvider
+  publicKey="your_public_api_key"
+  urlEndpoint="https://ik.imagekit.io/your_imagekit_id"
+  authenticator={()=>Promise} 
+  // This promise  resolves with an object containing the necessary security parameters i.e `signature`, `token`, and `expire`.
+  >
+  <IKUpload
+    onError={onError}
+    onSuccess={onSuccess}
+  />
+
+  // Passing different upload API options
+  <IKUpload
+    fileName="file-name.jpg"
+    tags={["sample-tag1", "sample-tag2"]}
+    customCoordinates={"10,10,10,10"}
+    isPrivateFile={false}
+    useUniqueFileName={true}
+    responseFields={["tags"]}
+    folder={"/sample-folder"}
+    ref={uploadRef}
+    onError={onError} onSuccess={onSuccess}
+    transformation={{
+      pre: 'l-text,i-Imagekit,fs-50,l-end', 
+      post: [
+        {
+          'type': 'transformation', 
+          'value': 'w-100'
+        }
+      ]
+    }}
+  />
+</ImageKitProvider>
+```
+
+## Demo application
+* The official step-by-step Next.js quick start guide - https://imagekit.io/docs/integration/next
+
+## Components
+
+The library includes 6 Components:
+
+* [`ImageKitProvider`](#ImageKitProvider) is used to define options such as `urlEndpoint`, `publicKey`, or `authenticator` for all child components. It does not render any UI elements.
```

Comment:
Why do we have two components which sort of does same thing i.e. `ImageKitProvider` and `ImageKitContext`

Reply:
provider and context are two different components, this change has been done to follow proper react naming conventions and follow the context provider pattern

Reply:
The provider is to define the values, the context is to consume the values defined values.